ambry-cloud/src/main/java/com/github/ambry/cloud/VcrReplicationManager.java

```diff
@@ -91,6 +112,30 @@ public void start() throws ReplicationException {
     vcrClusterParticipant.addListener(new VcrClusterParticipantListener() {
       @Override
       public void onPartitionAdded(PartitionId partitionId) {
+        if (partitionId.isEqual(cloudConfig.vcrHelixUpdaterParitionId)) {
+          vcrHelixUpdateLock.lock();
+          try {
+            if (!isAmbryListenerToUpdateVcrHelixRegistered) {
+              // Only register the listener once. Unfortunately, we can't unregister a listener, so we use
+              // isAmbryListenerToUpdateVcrHelixRegistered as the flag.
+              clusterMap.registerClusterMapListener(
+                  new com.github.ambry.cloud.VcrReplicationManager.AmbryListenerToUpdateVcrHelix());
```

Comment:
nit: is the fully specified classname needed here?

Reply:
fixed.